File: projects/rpn2tex/migrations_archive/java-module-by-module-1/src/main/java/com/rpn2tex/Token.java
package com.rpn2tex;

/**
 * Represents a token in an RPN expression.
 *
 * <p>Tokens are immutable objects that represent the smallest units of an RPN expression.
 * Each token has a type, optional value, and source location (line and column).
 *
 * <p>Example usage:
 * <pre>
 * Token numberToken = new Token(TokenType.NUMBER, "42", 1, 1);
 * Token plusToken = new Token(TokenType.PLUS, "+", 1, 4);
 * </pre>
 */
public final class Token {
    /** The type of this token. */
    public final TokenType type;

    /** The string value of this token (e.g., "42", "+"). */
    public final String value;

    /** The 1-based line number where this token appears. */
    public final int line;

    /** The 1-based column number where this token appears. */
    public final int column;

    /**
     * Creates a new token.
     *
     * @param type the type of token
     * @param value the string value of the token
     * @param line the 1-based line number
     * @param column the 1-based column number
     */
    public Token(TokenType type, String value, int line, int column) {
        this.type = type;
        this.value = value;
        this.line = line;
        this.column = column;
    }

    /**
     * Returns a string representation of this token for debugging.
     *
     * @return a string in the format "Token(TYPE, 'value', line:column)"
     */
    @Override
    public String toString() {
        return String.format("Token(%s, '%s', %d:%d)",
            type.name(), value, line, column);
    }
}

File: projects/rpn2tex/migrations/go-module-by-module-3/token.go
package rpn2tex

import "fmt"

// TokenType represents the type of a lexical token.
type TokenType int

const (
	// NUMBER represents a numeric literal token (integer or decimal).
	NUMBER TokenType = iota
	// PLUS represents the addition operator '+'.
	PLUS
	// MINUS represents the subtraction operator '-'.
	MINUS
	// MULT represents the multiplication operator '*'.
	MULT
	// DIV represents the division operator '/'.
	DIV
	// EOF represents the end of input.
	EOF
)

// String returns the string representation of a TokenType.
func (t TokenType) String() string {
	switch t {
	case NUMBER:
		return "NUMBER"
	case PLUS:
		return "PLUS"
	case MINUS:
		return "MINUS"
	case MULT:
		return "MULT"
	case DIV:
		return "DIV"
	case EOF:
		return "EOF"
	default:
		return fmt.Sprintf("TokenType(%d)", int(t))
	}
}

// Token represents a lexical token with its type, value, and source position.
// Tokens are immutable once created.
type Token struct {
	Type   TokenType // The type of the token
	Value  string    // The string value of the token
	Line   int       // Line number in source (1-based)
	Column int       // Column number in source (1-based)
}

// String returns a formatted string representation of the token.
// Format: Token(TYPE_NAME, 'value', line:column)
// This matches the Python __repr__ output for debugging purposes.
func (t Token) String() string {
	return fmt.Sprintf("Token(%s, '%s', %d:%d)", t.Type, t.Value, t.Line, t.Column)
}
